### ISSUE TITLE
Numerous fetching errors: berkeley-db, zlib, m4

### ISSUE THREAD

**Issue by zimaaaaa** [LLNL/spack#19584]
Hello! 

Please, watch below build details and errors.

### Steps to reproduce

```console
$ spack install gcc@7.3.0
```

### Error message

In the beginning it installed libsigsegv successfully, then:
```console
==> Installing berkeley-db
==> No binary for berkeley-db found: installing from source
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/0c/0cecb2ef0c67b166de93732769abdeba0555086d51de1090df325e18ee8da9c8.tar.gz
-=O=-      #    #    #    #                                                    
curl: (28) Resolving timed out after 10523 milliseconds
==> Fetching http://download.oracle.com/berkeley-db/db-18.1.40.tar.gz
-=O=-      #    #    #    #                                                    
curl: (28) Resolving timed out after 10523 milliseconds
==> Error: FetchError: All fetchers failed

/home/egor/soft/src/spack/lib/spack/spack/package.py:1270, in do_fetch:
       1267                                 self.spec.format('{name}{@version}'), ck_msg)
       1268
       1269        self.stage.create()
  >>   1270        err_msg = None if not self.manual_download else self.download_instr
       1271        self.stage.fetch(mirror_only, err_msg=err_msg)
       1272        self._fetch_time = time.time() - start_time
       1273


==> Warning: Skipping build of perl since berkeley-db failed
==> Warning: Skipping build of automake since perl failed
==> Warning: Skipping build of mpfr since automake failed
==> Warning: Skipping build of mpc since mpfr failed
==> Warning: Skipping build of gcc since mpc failed
==> Warning: Skipping build of gmp since automake failed
==> Warning: Skipping build of isl since gmp failed
==> Warning: Skipping build of autoconf since perl failed
```
After that it also installed pkgconf and then it fails zlib and m4:
```console
==> Installing zlib
==> No binary for zlib found: installing from source
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/c3/c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1.tar.gz
-=O=-      #    #    #    #                                                    
curl: (28) Resolving timed out after 10519 milliseconds
==> Fetching http://zlib.net/fossils/zlib-1.2.11.tar.gz
-=O=-      #    #    #    #                                                    
curl: (28) Resolving timed out after 10520 milliseconds
==> Error: FetchError: All fetchers failed

/home/egor/soft/src/spack/lib/spack/spack/package.py:1270, in do_fetch:
       1267                                 self.spec.format('{name}{@version}'), ck_msg)
       1268
       1269        self.stage.create()
  >>   1270        err_msg = None if not self.manual_download else self.download_instr
       1271        self.stage.fetch(mirror_only, err_msg=err_msg)
       1272        self._fetch_time = time.time() - start_time
       1273


==> Installing m4
==> No binary for m4 found: installing from source
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/ab/ab2633921a5cd38e48797bf5521ad259bdc4b979078034a3b790d7fec5493fab.tar.gz
-=O=-      #    #    #    #                                                    
curl: (28) Resolving timed out after 10521 milliseconds
==> Fetching https://ftpmirror.gnu.org/m4/m4-1.4.18.tar.gz
-=O=-              #    #     #     #                                          
curl: (28) Resolving timed out after 10521 milliseconds
==> Fetching https://ftp.gnu.org/gnu/m4/m4-1.4.18.tar.gz
######################################################################### 100.0%
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/fc/fc9b61654a3ba1a8d6cd78ce087e7c96366c290bc8d2c299f09828d793b853c8
-=O=-      #    #    #    #                                                    
curl: (28) Resolving timed out after 10522 milliseconds
==> Fetching https://src.fedoraproject.org/rpms/m4/raw/5d147168d4b93f38a4833f5dd1d650ad88af5a8a/f/m4-1.4.18-glibc-change-work-around.patch
-=O=-      #    #    #    #                                                    
curl: (28) Resolving timed out after 10519 milliseconds
==> Error: FetchError: All fetchers failed

/home/egor/soft/src/spack/lib/spack/spack/package.py:1279, in do_fetch:
       1276
       1277        self.stage.cache_local()
       1278
  >>   1279        for patch in self.spec.patches:
       1280            patch.fetch()
       1281            if patch.stage:
       1282                patch.stage.cache_local()


==> Warning: Skipping build of libtool since m4 failed
```
ncurses readline and gbdm are in success, then it makes the summary:
```console
==> Error: Installation of gcc failed.  Review log for details
```

My OS: Ubuntu 18.04. Thanks for the help in advance and sorry for probable difficulties with understanding, i'm beginner with spack in particular and building programs in general.

All the best,
Egor

**Comment by adamjstewart** [comment]
Can you run `spack fetch zlib` a few times and see if it eventually succeeds? I'm not sure why so many of your downloads are timing out, they work fine for me. Are you behind a particularly strict firewall?

**Comment by zimaaaaa** [comment]
Firstly I tried to run `spack fetch zlib` a few times, but nothing changes at all. Then I checked my router settings and there was no any firewall setups but I tried to change my internet source from one ISP (for wired connection) to another (for wireless mobile connection) and after that all fetchings were successfull. So I think this issue was actually in internet-provider's firewall. Thank you very much for your help!
